### PR TITLE
prepare space for fused docs

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Make a dummy doc and set up related CI/CD jobs...
For preserving continuity, we use symlinks to the existing pages. When all is settled up, we will project the real pages via link to the new location and sunset the old docs location.
The cleaning requirements are also separated in #14216

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

This is a preparation stage for docs unification. If you want to edit/update docs for ongoing documentation, do it in `source-app` or `source-pytorch`. Otherwise, if the changes are related to the future/fused docs, please rename/move the existing symlink or make a new file where you can still copy-paste the page content...

cc @borda

### CODE REVIEW

**Comment by Borda** [comment]
ok, one remaining issue lest is with unification of templates:
```
Extension error (pt_lightning_sphinx_theme.extensions.lightning):
Handler <function process_card_nodes at 0x7f0acfce40d0> for event 'doctree-resolved' threw an exception (exception: 'BuildEnvironment' object has no attribute 'all_cardnodes')
make: *** [Makefile:19: html] Error 2
```

**Review by awaelchli — APPROVED** [review]

**Review by Felonious-Spellfire — APPROVED** [review]

**Review by akihironitta — COMMENTED** [review]
(I guess this has been discussed already somewhere and/or can be discussed/polished later in follow-up PRs, but)

IMHO, having both app and pl at the same directory level seems confusing, e.g.
```
docs/source-lit/accelerators  # <- pl
docs/source-lit/strategies    # <- pl
...
docs/source-lit/code_samples  # <- app
docs/source-lit/workflows     # <- app
...
```

**Comment by Borda** [comment]
> IMHO, mixing both app and pl and having both at the same directory level seems confusing, e.g.

This is just init state, and all shall be restructured by @Felonious-Spellfire in following PRs

File: requirements/lit/docs.txt
Comment (yurijmikhalevich) [inline requirements/lit/docs.txt:7]: This will tell it to use the "Pytorch Lightning" theme (without the lightning.ai header) and should unblock this PR. We should update this to use the theme with the lightning.ai header before merging.

```suggestion
https://github.com/PyTorchLightning/lightning_sphinx_theme/archive/master.zip#egg=pt-lightning-sphinx-theme
```

**Review by akihironitta — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]

File: requirements/lit/docs.txt
Comment (yurijmikhalevich) [inline requirements/lit/docs.txt:7]: The theme was installed correctly. We install the same theme the same way for PL docs.

File: requirements/lit/docs.txt
Comment (yurijmikhalevich) [inline requirements/lit/docs.txt:9]: What is this supposed to do?

File: requirements/lit/docs.txt
Comment (rohitgr7) [reply requirements/lit/docs.txt:7]: yep, it's the same but wasn't the same way. But that doesn't matter.

File: requirements/lit/docs.txt
Comment (rohitgr7) [reply requirements/lit/docs.txt:9]: install packages used to build tutorials: https://pytorch-lightning.readthedocs.io/en/latest/tutorials.html?highlight=tutorials

**Pull request merged**
